rust/vapid/src/lib.rs

```diff
@@ -428,37 +439,41 @@ mod tests {
         // And verify that the signature works.
         // we do integration verify in `test_verify`
         verify(vresult).expect("Signed claims failed to self verify");
-
     }
 
     // TODO: Test fail cases, verification, values
 
     #[test]
-    fn test_sign_bad_sub(){
-        let key = Key::generate();
+    fn test_sign_bad_sub() {
+        let key = Key::generate().unwrap();
         let mut claims: HashMap<String, serde_json::Value> = HashMap::new();
-        claims.insert("sub".into(), serde_json::Value::from(String::from("invalid")));
+        claims.insert(
+            "sub".into(),
+            serde_json::Value::from(String::from("invalid")),
+        );
         match sign(key, &mut claims) {
```

Comment:
nit: `unwrap_err/expect_err` are great for cases like these in tests

```suggestion
        let err = sign(key, &mut claims).expect_err("Failed to reject invalid sub");
        let errstr = format!("{:?}", err);
        assert!(errstr.contains("not a valid HTML reference"));
```

Reply:
True, but I think I want to spell out exactly what I'm looking for. Having the two part match makes it a bit clearer to me, at least.